Ruby/wc/ccwc.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative './file_content'
+
+def find_input_command(input_array)
+  return '-c' if input_array.include?('-c')
+  return '-l' if input_array.include?('-l')
+  return '-w' if input_array.include?('-w')
+  return '-m' if input_array.include?('-m')
+
+  'invalid'
+end
+
+def print_info(count, file_name)
+  puts "#{count} #{file_name}" unless file_name.nil?
+  puts count if file_name.nil?
+end
+
+def find_index_of_file_and_command(input_array)
+  input = find_input_command(input_array[0])
+  return [input, 1] if input != 'invalid'
+
+  input = find_input_command(input_array[1])
+  [input, 0] if input != 'invalid'
+end
+
+def read(file)
+  file_content = ''
+  File.foreach(file) do |line|
+    file_content += line
+  end
+  file_content
+end
+
+input = ''
+# `ARGV` is a special global variable that holds the command-line arguments passed to a Ruby script.
+input_array = ARGV
+command_index = find_index_of_file_and_command(input_array)
+if !$stdin.tty?
+  # STDIN.tty? to check if the standard input is interactively coming from the terminal.
+  # eg: `cat test.txt | ruby ccwb.rb -c`
+  # in the above input - test.txt is not being directly used as the standard input to the ruby script,
+  # rather the contents of the file are used an input.
+  #
+  # `$stdin` is a global variable that represents the standard input stream.
+  # The use of $stdin in this context allows the script to read from the standard input,
+  # which is being provided through the pipe (|) from the cat command.
+  file_content = ''
+  $stdin.each_line do |line|
+    file_content += line
+  end
+  input = command_index[0]
+elsif input_array.length > 1
+  input = command_index[0]
+  file_name = input_array[command_index[1]]
+  file_content = read(file_name)
+elsif input_array.length.positive?
+  file_name = input_array[0]
+  file_content = read(file_name)
+end
```

Comment:
There's no `else` here - any chance none of these conditions are true? 🤔

Reply:
actually, there's a chance. In that case, the exception has to be handled.

Reply:
addressed it here - https://github.com/aganapuram/CodingChallengesJC/pull/1/commits/c35fdc9a57a9b0c5a0b70433fb6cad5c05eabafa